searchcore/src/vespa/searchcore/proton/documentmetastore/lid_allocator.cpp

```diff
@@ -27,11 +27,19 @@ LidAllocator::LidAllocator(uint32_t size,
       _freeLids(size, capacity, genHolder, true, false),
       _usedLids(size, capacity, genHolder, false, true),
       _pendingHoldLids(size, capacity, genHolder, false, false),
-      _lidFreeListConstructed(false),
       _activeLids(size, capacity, genHolder, false, false),
-      _numActiveLids(0u)
-{
+      _numActiveLids(0u),
+      _lidFreeListConstructed(false)
+{ }
 
+vespalib::MemoryUsage
+LidAllocator::getMemoryUsage() const {
+    vespalib::MemoryUsage usage;
+    size_t allocated = sizeof(*this) + _freeLids.byteSize() + _usedLids.byteSize() +
```

Comment:
Calling byteSize() on a LidStateVector is unsafe unless the proper generation guard is held if called from a reader thread.

Reply:
It is called from the same place as used to call getUsedLidsSize(), updateStat().
That should be fine as it happens as part of writer thread.